tests/sparseml/pytorch/sparsification/distillation/test_per_layer_distillation.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Callable
+
+import pytest
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from sparseml.pytorch.sparsification import (
+    DistillationModifier,
+    Modifier,
+    ScheduledModifier,
+)
+from tests.sparseml.pytorch.helpers import LinearNet, create_optim_sgd
+from tests.sparseml.pytorch.sparsification.test_modifier import ScheduledModifierTest
+
+
+from tests.sparseml.pytorch.helpers import (  # noqa isort:skip
+    test_epoch,
+    test_loss,
+    test_steps_per_epoch,
+)
+
+
+DISTILLATION_MODIFIERS = [
+    lambda: DistillationModifier(start_epoch=0.0),
```

Comment:
Probably want:
1. `PerLayerDistillationModifier` with no projections
2. `PerLayerDistillationModifier` with mixed projections (some layers projected, some not)